openssl/evp.go

```diff
@@ -393,3 +393,34 @@ func newEvpFromParams(id C.int, selection C.int, params []C.OSSL_PARAM) (C.GO_EV
 	}
 	return pkey, nil
 }
+
+// encodeEcPoint encodes pt.
+func encodeEcPoint(group C.GO_EC_GROUP_PTR, pt C.GO_EC_POINT_PTR) ([]byte, error) {
+	// Get encoded point size.
+	n := C.go_openssl_EC_POINT_point2oct(group, pt, C.GO_POINT_CONVERSION_UNCOMPRESSED, nil, 0, nil)
+	if n == 0 {
+		return nil, newOpenSSLError("EC_POINT_point2oct")
+	}
+	// Encode point into bytes.
+	bytes := make([]byte, n)
+	n = C.go_openssl_EC_POINT_point2oct(group, pt, C.GO_POINT_CONVERSION_UNCOMPRESSED, base(bytes), n, nil)
+	if n == 0 {
+		return nil, newOpenSSLError("EC_POINT_point2oct")
+	}
+	return bytes, nil
+}
+
+// generateAndEncodePublicKey calls newPubKeyPointFn to generate a public key point and then encodes it.
+func generateAndEncodePublicKey(nid C.int, newPubKeyPointFn func(group C.GO_EC_GROUP_PTR) (C.GO_EC_POINT_PTR, error)) ([]byte, error) {
```

Comment:
Is this the right place to add these couple of functions? They are not using `EVP_*` functions nor being used by `evp.go`.

Reply:
Agree. I've moved these functions, and a couple others, to the new `ec.go` file.